src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -569,5 +569,12 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
                 dump.Append('|');
             }
         }
+
+        public static string DecodeString(IByteBuffer src, int readerIndex, int len, Encoding encoding)
+        {
+            if (len == 0) return string.Empty;
+            
+            return encoding.GetString(src.Array.Slice(len));
```

Comment:
buffer may not have a backing array (i.e. `buf.HasArray == false`). CompositeByteBuffer is a good example.